packages/blockchain/src/index.ts

```diff
@@ -939,30 +943,48 @@ export default class Blockchain implements BlockchainInterface {
         }
       }
 
-      // set total difficulty in the current context scope
-      if (this._headHeaderHash) {
-        currentTd.header = await this.getTotalDifficulty(this._headHeaderHash)
+      if (block._common.consensusType() === ConsensusType.ProofOfStake) {
+        if (block._common.hardfork() !== 'merge') {
```

Comment:
what is this if statement trying to check? as noted in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1406/files#r689925035, consensusType() changes to ProofOfStake after arriving at the merge hardfork block, so this added condition is redundant

Reply:
Yes, also here: the PoS check is enough. We likely also (a bit like the Block library) want to have the Blockchain library as "Merge" HF independent as possible, so if things are on `ProofOfStake`, the blockchain should behave like a PoS blockchain, no matter where this is coming from (from a Merge HF, from another HF switching to PoS, from Genesis).

The `Hardfork.Merge` HF should *only* be used if things *only* happen right along this switch (so e.g. if there would be something in the spec like `when the Merge HF occur put "We made it!" in the `extraData` field of the block 😋).